src/core/stream_send.c

```diff
@@ -234,6 +234,26 @@ QuicStreamSendShutdown(
                 QUIC_STREAM_SEND_FLAG_OPEN |
                 QUIC_STREAM_SEND_FLAG_FIN);
         }
+    } else {
+        uint64_t CurrOffset = Stream->UnAckedOffset;
```

Comment:
If `Stream->Flags.LocalCloseReset` is already set, then we've already sent a (non-reliable) reset, and there is no need to do anything more.

Reply:
Added a check! Also checked for graceful closes as well.